src/features/ISECompatibility.ts

```diff
@@ -0,0 +1,52 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+import * as vscode from "vscode";
+import { LanguageClient } from "vscode-languageclient";
+import { IFeature } from "../feature";
+
+/**
+ * A feature to implement commands to make code like the ISE and reset the settings.
+ */
+export class ISECompatibilityFeature implements IFeature {
+    private iseCommand: vscode.Disposable;
+    private defaultCommand: vscode.Disposable;
+    private settings = [
+        {section: "workbench.activityBar", setting: "visible", value: false},
+        {section: "debug", setting: "openDebug", value: "neverOpen"},
+        {section: "editor", setting: "tabCompletion", value: "on"},
+        {section: "powershell.integratedConsole", setting: "focusConsoleOnExecute", value: false},
+        {section: "powershell.integratedConsole", setting: "showOnStartup", value: false},
+        {section: "files", setting: "defaultLanguage", value: "powershell"},
+        {section: "workbench", setting: "colorTheme", value: "PowerShell ISE"},
+    ];
+    private languageClient: LanguageClient;
+
+    constructor() {
+        this.iseCommand = vscode.commands.registerCommand("PowerShell.SetISECompatibility",
+            () => this.SetISECompatibility());
+        this.defaultCommand = vscode.commands.registerCommand("PowerShell.UnsetISECompatibility",
+            () => this.UnsetISECompatibility());
+    }
+
+    public dispose() {
+        this.iseCommand.dispose();
+        this.defaultCommand.dispose();
+    }
+
+    public setLanguageClient(languageclient: LanguageClient) {
+        this.languageClient = languageclient;
+    }
+
+    private SetISECompatibility() {
+        this.settings.forEach((value) => {
+            vscode.workspace.getConfiguration(value.section).update(value.setting, value.value, true);
+        });
+    }
+
+    private UnsetISECompatibility() {
+        this.settings.forEach((value) => {
+            vscode.workspace.getConfiguration(value.section).update(value.setting, undefined, true);
```

Comment:
We should do like
```ts
if (/* setting is set to ISE value */) {
    // set setting to undefined
} 
```

That way we don't mess up people's settings if they changed a setting after the fact to something that's not an ISE mode value.

Reply:
The perfect example is the theme. If I want ISE Mode, but use a different theme, I don't want this to affect that.

Reply:
> The perfect example is the theme. If I want ISE Mode, but then after enabling changed to use a different theme, I don't want this to affect that.

This brings up an interesting thought... Do we want to store the theme when they run enable and if it's still ISE theme when they disable, set it back to their previous choice? Or does this introduce unneeded complexity?